core/java/iesi-core/src/test/java/io/metadew/iesi/common/crypto/FrameworkCryptoTest.java

```diff
@@ -39,14 +40,15 @@ public void testJavaKeystoreWrongPassword() {
         try {
             new JavaKeystore().loadKey(userinput.toCharArray(), keystoreLocation, alias);
         } catch (Exception e) {
-            assertEquals("java.io.IOException: Integrity check failed: java.security.UnrecoverableKeyException: Failed PKCS12 integrity checking", e.getMessage());
+            assertThat("java.io.IOException: Integrity check failed: java.security.UnrecoverableKeyException: Failed PKCS12 integrity checking").isEqualTo(e.getMessage());
         }
     }
 
     @Test
     public void testKeyInConf() {
-        if (Configuration.getInstance().getProperty("iesi.security.encryption.key").isPresent()) {
-            assertEquals("c7c1e47391154a6a", Configuration.getInstance().getMandatoryProperty("iesi.security.encryption.key").toString());
+        if (!Configuration.getInstance().getProperty("iesi.security.encryption.key").isPresent()) {
+            assertThat("y8c1e47391154a6c").isEqualTo(Configuration.getInstance().getMandatoryProperty("iesi.security.encryption.key").toString());
+            assertThat("y8c1e47391154a6b").isEqualTo(not(Configuration.getInstance().getMandatoryProperty("iesi.security.encryption.key").toString()));
```

Comment:
A test should not contain an if statement. You should delete this if statement.
Also, the goal of the test is to verify the encryption key used by FrameworkCrypto, not the key stored in configuration.
You will probably have to mock/spy the Configuration singleton to manipulate this behaviour